src/keychain_mac.cpp

```diff
@@ -111,28 +108,25 @@ void updateError(keychain::Error &err, OSStatus status) {
  */
 OSStatus modifyPassword(const std::string &serviceName, const std::string &user,
                         const std::string &password) {
-    SecKeychainItemRef item = NULL;
-    OSStatus status = SecKeychainFindGenericPassword(
-        defaultUserKeychain,
-        static_cast<UInt32>(serviceName.length()),
-        serviceName.data(),
-        static_cast<UInt32>(user.length()),
-        user.data(),
-        NULL, // unused output parameter
-        NULL, // unused output parameter
-        &item);
+    CFStringRef cfServiceName = CFStringCreateWithCString(NULL, serviceName.c_str(), kCFStringEncodingUTF8);
```

Comment:
This and other `CF*Create*` will apparently return NULL if they fail. The documentation only says this happens "if there was a problem creating the object". I imagine encoding errors or out-of-memory could lead to this.

In any case, I think we should handle these. In terms of `keychain::Error` I'd just make it a `GenericError`, possibly with a generic message like "Object creation failed".

`modifyPassword` here is a bit annoying because it wants to return an `OSStatus`, which we don't get here. I'm tempted to say we should remove `modifyPassword` and inline it in `setPassword`. Readability suffers in this case, but we could get rid of some code duplication in turn, as I suggested below.